### PR TITLE
[tre] Add CMake config files.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] SHA512s are updated for each updated download
- [ ] The "supports" clause reflects platforms that may be fixed by this new version
- [ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [ ] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by dg0yt — COMMENTED** [review]
IMO the port doesn't lack unofficial CMake config, but official pc files.
A frequent problem for ports which don't use the official build system. vcpkg.

File: ports/tre/portfile.cmake [hunk 1]
```diff
@@ -13,13 +13,17 @@ if(VCPKG_TARGET_IS_MINGW)
...
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
 )
 
 vcpkg_cmake_install()
+vcpkg_copy_pdbs()
+vcpkg_cmake_config_fixup(CONFIG_PATH share/unofficial-${PORT})
```
Comment (LilyWangLL) [inline ports/tre/portfile.cmake:24]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-${PORT} CONFIG_PATH share/unofficial-${PORT})
```

File: ports/tre/CMakeLists.txt [hunk 2]
```diff
@@ -32,9 +32,21 @@ add_library(tre ${SRCS})
 
 install(
```
Comment (LilyWangLL) [inline ports/tre/CMakeLists.txt:33]:
```suggestion
target_include_directories(tre PUBLIC "$<INSTALL_INTERFACE:include>")

install(
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/tre/portfile.cmake (hunk 1 above)
Comment (dg0yt) [inline ports/tre/portfile.cmake:24]: Good spot, but 
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-${PORT})
```

**Comment by teo-tsirpanis** [comment]
Feedback addressed.

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

**Review by LilyWangLL — APPROVED** [review]

**Comment by teo-tsirpanis** [comment]
Can it be merged?

**Comment by LilyWangLL** [comment]
> Can it be merged?

Pinging @dan-shaw for response, this PR is ready to merge.

**Pull request merged**